regression/rt_ar_react_1x2v_quadraticV.c

```diff
@@ -0,0 +1,969 @@
+    .vpar_max_ion = vpar_max_ion, 
+    .mu_max_ion = mu_max_ion,
+    .vpar_max_Ar = vpar_max_Ar, 
+    .mu_max_Ar = mu_max_Ar, 
+    .finalTime = finalTime, 
+    .numFrames = numFrames,
+  };
+  return ctx;
+}
+
+struct all_of_1species_reactions {
+  struct gkyl_gyrokinetic_react_type recombination[GKYL_MAX_SPECIES];
+  struct gkyl_gyrokinetic_react_type ionization[GKYL_MAX_SPECIES];
+};
+
+struct all_of_1species_reactions*
+create_isonuclear_react(double ion_mass, double electron_mass, int max_z, enum gkyl_ion_type ion_id, char* electron_name, char* ion_name)
+{
+  static struct all_of_1species_reactions reactions[4];  // index = gkyl_react_self_type
+  // 0 = elc, 1=ion, 2 = donor, 3 = recvr
+  char ion_nm[128];
+  char donor_nm[128];  // Same as recvr_nm
+  for (int j=0; j<4; j++) {
+    for (int i=0; i<max_z; i++) {
+      snprintf(ion_nm, sizeof(ion_nm), "%s%d", ion_name, i+1);
+      snprintf(donor_nm, sizeof(donor_nm), "%s%d", ion_name, i);
+      reactions[j].recombination[i].react_id = GKYL_REACT_RECOMB;
+      reactions[j].recombination[i].ion_id = ion_id;
+      strcpy(reactions[j].recombination[i].elc_nm, electron_name);
+      reactions[j].recombination[i].charge_state = i;
+      reactions[j].recombination[i].ion_mass = ion_mass;
+      reactions[j].recombination[i].elc_mass = electron_mass;
+      strcpy(reactions[j].recombination[i].ion_nm, ion_nm);
+      strcpy(reactions[j].recombination[i].recvr_nm, donor_nm);
+      reactions[j].recombination[i].type_self = j;
+      
+      reactions[j].ionization[i].react_id = GKYL_REACT_IZ;
+      reactions[j].ionization[i].ion_id = ion_id;
+      strcpy(reactions[j].ionization[i].elc_nm, electron_name);
+      reactions[j].ionization[i].charge_state = i;
+      reactions[j].ionization[i].ion_mass = ion_mass;
+      reactions[j].ionization[i].elc_mass = electron_mass;
+      strcpy(reactions[j].ionization[i].ion_nm, ion_nm);
+      strcpy(reactions[j].ionization[i].donor_nm, donor_nm);
+      reactions[j].ionization[i].type_self = j;
+    }
+  }
+  return reactions;
+}
+
+void
+write_data(struct gkyl_tm_trigger *iot, gkyl_gyrokinetic_app *app, double tcurr)
+{
+  if (gkyl_tm_trigger_check_and_bump(iot, tcurr)) {
+    gkyl_gyrokinetic_app_write(app, tcurr, iot->curr-1);
+    gkyl_gyrokinetic_app_calc_mom(app); gkyl_gyrokinetic_app_write_mom(app, tcurr, iot->curr-1);
+  }
+}
+
+int  
+main(int argc, char **argv)
+{
+  struct gkyl_app_args app_args = parse_app_args(argc, argv);
+
+  if (app_args.trace_mem) {
+    gkyl_cu_dev_mem_debug_set(true);
+    gkyl_mem_debug_set(true);
+  }
+
+  struct gk_app_ctx ctx = create_ctx(); // context for init functions
+
+  // Reactions for argon isonuclear sequence
+  struct all_of_1species_reactions *reactions = create_isonuclear_react(ctx.massAr, ctx.massElc, 18, GKYL_ION_AR, "elc", "Ar");
+
+  int NX = APP_ARGS_CHOOSE(app_args.xcells[0], 2);
+  int NV = APP_ARGS_CHOOSE(app_args.vcells[0], 6);
+  int NMU = APP_ARGS_CHOOSE(app_args.vcells[1], 4);
```

Comment:
@jRoeltgen please see how number of cells are specified and passed to the app in established tests (e.g. rt_gk_sheath_3x2v_p1) and stick to that